conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt

```diff
@@ -0,0 +1,923 @@
+    int:Signed16? d = -454;
+    int:Signed8? e = 100;
+
+    int:Unsigned32? f = 21474836;
+    int:Unsigned16? g = 54522;
+    int:Unsigned8? h = 255;
+
+    int? i = ();
+    byte? j = ();
+    int:Signed32? k = ();
+    int:Unsigned16? l = ();
+
+    io:println((a & i).toBalString()); // @output ()
+    io:println((a & j).toBalString()); // @output ()
+    io:println((k & a).toBalString()); // @output ()
+    io:println((l & a).toBalString()); // @output ()
+
+    io:println((b & i).toBalString()); // @output ()
+    io:println((b & j).toBalString()); // @output ()
+    io:println((b & k).toBalString()); // @output ()
+    io:println((b & l).toBalString()); // @output ()
+
+    io:println((c & i).toBalString()); // @output ()
+    io:println((c & j).toBalString()); // @output ()
+    io:println((k & c).toBalString()); // @output ()
+    io:println((c & l).toBalString()); // @output ()
+
+    io:println((d & i).toBalString()); // @output ()
+    io:println((d & j).toBalString()); // @output ()
+    io:println((d & k).toBalString()); // @output ()
+    io:println((d & l).toBalString()); // @output ()
+
+    io:println((e & i).toBalString()); // @output ()
+    io:println((e & j).toBalString()); // @output ()
+    io:println((e & k).toBalString()); // @output ()
+    io:println((e & l).toBalString()); // @output ()
+
+    io:println((f & i).toBalString()); // @output ()
+    io:println((f & j).toBalString()); // @output ()
+    io:println((f & k).toBalString()); // @output ()
+    io:println((f & l).toBalString()); // @output ()
+
+    io:println((g & i).toBalString()); // @output ()
+    io:println((g & j).toBalString()); // @output ()
+    io:println((g & k).toBalString()); // @output ()
+    io:println((g & l).toBalString()); // @output ()
+
+    io:println((h & i).toBalString()); // @output ()
+    io:println((h & j).toBalString()); // @output ()
+    io:println((h & k).toBalString()); // @output ()
+    io:println((h & l).toBalString()); // @output ()
+
+    io:println((i & i).toBalString()); // @output ()
+    io:println((i & j).toBalString()); // @output ()
+    io:println((i & k).toBalString()); // @output ()
+    io:println((i & l).toBalString()); // @output ()
+
+    io:println((j & j).toBalString()); // @output ()
+    io:println((j & k).toBalString()); // @output ()
+    io:println((j & l).toBalString()); // @output ()
+
+    io:println((k & k).toBalString()); // @output ()
+    io:println((k & l).toBalString()); // @output ()
+
+    io:println((l & l).toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test bitwise and expression where the operands are user defined type T where the T is a subtype of int.
+Label: binary-bitwise-expr, module-type-defn
+
+type MyType int;
+
+public function main() {
+    MyType a = 5;
+    MyType b = 433;
+    int c = 765;
+
+    printResult(a & b); // @output 1
+    printResult(a & 12345); // @output 1
+    printResult(b & c); // @output 177
+    printResult(a & c); // @output 5
```

Comment:
Shall we also add some tests with user-defined types such as `type MyType int:Signed16|int:Unsigned32` and unions like `int:Signed16|int:Unsigned32`? Check with nillable types and other `binary-bitwise-expr` as well.

Reply:
Added